File: generate-icons.py
#!/usr/bin/env python3
"""Generate PNG icons from SVG with proper aspect ratio handling."""

import os
from pathlib import Path

try:
    from PIL import Image
    import cairosvg
except ImportError:
    print("Installing required packages...")
    import subprocess
    subprocess.check_call(['pip', 'install', '--user', 'pillow', 'cairosvg'])
    from PIL import Image
    import cairosvg

# Paths
SCRIPT_DIR = Path(__file__).parent
SVG_PATH = SCRIPT_DIR / 'icons' / 'icon.svg'
SIZES = [16, 48, 128]

def generate_icons():
    """Generate PNG icons from SVG with transparent background."""
    for size in SIZES:
        output_path = SCRIPT_DIR / 'icons' / f'icon{size}.png'
        
        # Convert SVG to PNG with proper aspect ratio (fit contain)
        png_data = cairosvg.svg2png(
            url=str(SVG_PATH),
            output_width=size,
            output_height=size
        )
        
        # Open with PIL to ensure transparency
        img = Image.open(io.BytesIO(png_data))
        
        # Save with transparency
        img.save(output_path, 'PNG')
        print(f'Generated {output_path}')
    
    print('All icons generated successfully!')

if __name__ == '__main__':
    import io
    generate_icons()

File: .stryker-tmp/sandbox-6SwJaH/generate-icons.js
// @ts-nocheck
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const svgPath = path.join(__dirname, 'icons', 'icon.svg');
const sizes = [16, 48, 128];

async function generateIcons() {
  const svgBuffer = fs.readFileSync(svgPath);

  for (const size of sizes) {
    const outputPath = path.join(__dirname, 'icons', `icon${size}.png`);

    await sharp(svgBuffer)
      .resize(size, size, {
        fit: 'contain',
        background: { r: 0, g: 0, b: 0, alpha: 0 }
      })
      .png()
      .toFile(outputPath);

    console.log(`Generated ${outputPath}`);
  }

  console.log('All icons generated successfully!');
}

generateIcons().catch(console.error);
